tests/test_build_changes.py

```diff
@@ -32,6 +38,45 @@ def test_build(app):
         ' Can now be replaced with a different allocator.</a>')
     assert malloc_html in htmltext
 
+    assert '[new config] 4 added' in app._status.getvalue()
+
+    # config changed (due to how the fixture works, changed master_doc)
+    args, kwargs = app_params
+    app0 = make_app(*args, **kwargs)
+    assert app0.env.config_status == CONFIG_CHANGED
+    app0.build()
+    assert "[config changed ('master_doc')] 4 added" in app0._status.getvalue()
+
+    # no changes
+    confoverrides = dict(master_doc=app0.env.config['master_doc'])
+    app1 = make_app(*args, confoverrides=confoverrides, **kwargs)
+    assert app0.env.config_status == CONFIG_OK
+    app1.build()
+    assert "0 added, 0 changed, 0 removed" in app1._status.getvalue()
+
+    # change a file
+    fname = os.path.join(app.srcdir, confoverrides['master_doc'] + '.rst')
```

Comment:
Please use `index.rst` directly. I think concreteness increases readability of testcase.